### ISSUE TITLE
Can't SELECT the mysql.user table in mariadb 10.4 or later.

### ISSUE THREAD

**Issue by ggreenworld** [datacharmer/dbdeployer#120]
**Describe the bug**

On mariadb 10.4 or later, when I SELECT the mysql.user table, I get the following error.

```
ERROR 1449 (HY000): The user specified as a definer ('mariadb.sys'@'localhost') does not exist
```

**To Reproduce**

##### Example for version 10.5.6
Immediately after unpacking and deploying the following tarball, we will run the query.

https://downloads.mariadb.com/MariaDB/mariadb-10.5/bintar-linux-x86_64/mariadb-10.5.6-linux-x86_64.tar.gz

```
mysql [localhost:10506] {root} ((none)) > select user,host from mysql.user;
ERROR 1449 (HY000): The user specified as a definer ('mariadb.sys'@'localhost') does not exist
```

##### Example for version 10.4.15

https://downloads.mariadb.com/MariaDB/mariadb-10.4/bintar-linux-x86_64/mariadb-10.4.15-linux-x86_64.tar.gz

```
mysql [localhost:10415] {root} ((none)) > select user,host from mysql.user;
ERROR 1449 (HY000): The user specified as a definer ('mariadb.sys'@'localhost') does not exist
```

##### Example for version 10.3.25

I was able to successfully SELECT.

https://downloads.mariadb.com/MariaDB/mariadb-10.3/bintar-linux-x86_64/mariadb-10.3.25-linux-x86_64.tar.gz
```
mysql [localhost:10325] {root} ((none)) > select user,host from mysql.user;
+-------------+-----------+
| user        | host      |
+-------------+-----------+
| msandbox    | 127.%     |
| msandbox_ro | 127.%     |
| msandbox_rw | 127.%     |
| rsandbox    | 127.%     |
| msandbox    | localhost |
| msandbox_ro | localhost |
| msandbox_rw | localhost |
| root        | localhost |
+-------------+-----------+
8 rows in set (0.001 sec)
```

**Expected behavior**

With VM(CentOS8) of LXC, I was able to query successfully with MariaDB 10.5.6 which I installed RPM (from the repository).
(`mariadb105` is VM hostname.)
```
MariaDB [(none)]> select user,host from mysql.user;
+-------------+------------+
| User        | Host       |
+-------------+------------+
|             | localhost  |
| mariadb.sys | localhost  |
| mysql       | localhost  |
| root        | localhost  |
|             | mariadb105 |
+-------------+------------+
5 rows in set (0.005 sec)
```

I think this is because `mariadb.sys@localhost` user does not exist, as we can see from the error message.

https://jira.mariadb.org/browse/MDEV-22909

If I check mysql.global_priv in an instance of the dbdeployer environment, the user certainly does not exist.

```
mysql [localhost:10506] {root} ((none)) > select user,host from mysql.global_priv;
+-------------+-----------+
| user        | host      |
+-------------+-----------+
| msandbox    | 127.%     |
| msandbox_ro | 127.%     |
| msandbox_rw | 127.%     |
| rsandbox    | 127.%     |
| msandbox    | localhost |
| msandbox_ro | localhost |
| msandbox_rw | localhost |
| root        | localhost |
+-------------+-----------+
8 rows in set (0.000 sec)
```

**Environment:**
 - OS: Ubuntu 18.04 on WSL version2
 - dbdeployer version: 1.55.0
 - tarball full name (e.g. mysql-5.6.40-linux-glibc2.12-x86_64.tar.gz)
   (listed above.)
 - tarball origin (from where it was downloaded) 
   (listed above.)

**Comment by datacharmer** [comment]
Thanks for reporting this issue. I was not aware that MariaDB had added a specific user for system queries, like MySQL did some time ago. I know what the problem is and will publish a fix with the next release.

**Comment by datacharmer** [comment]
Fixed in [version 1.56.0](https://github.com/datacharmer/dbdeployer/releases/tag/v1.56.0)

**Comment by ggreenworld** [comment]
@datacharmer 
Thank you for fixing it.
